### PR TITLE
Roles on contributor tags

### PR DESCRIPTION
**Pull request by jamieparkinson**
Merge https://github.com/wellcomecollection/wellcomecollection.org/pull/5884 first!

Also includes some typescript conversions for `WorkDetails*` components
Resolves https://github.com/wellcomecollection/wellcomecollection.org/issues/5758

### CODE REVIEW

File: catalogue/webapp/components/WorkDetailsProperty/WorkDetailsProperty.tsx [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
-import { type Node } from 'react';
 import { font } from '@weco/common/utils/classnames';
 import SpacingComponent from '@weco/common/views/components/SpacingComponent/SpacingComponent';
+import { FunctionComponent, ReactNode } from 'react';
 
-type Props = {| title: ?string, children: Node |};
+type Props = { title?: string; children: ReactNode };
 
-const WorkDetailsProperty = ({ title, children }: Props) => {
+const WorkDetailsProperty: FunctionComponent<Props> = ({
```
Comment (jamesgorrie) [inline catalogue/webapp/components/WorkDetailsProperty/WorkDetailsProperty.tsx:7]: You can also use the `PropsWithChildren` type.

Comment (jamieparkinson) [reply catalogue/webapp/components/WorkDetailsProperty/WorkDetailsProperty.tsx:7]: If you work this out I'd be grateful - you [shouldn't](https://www.carlrippon.com/react-children-with-typescript/) actually need anything other than `FunctionComponent<Props>` as that does the injection of children itself (have a look in its definition), but doing that and removing the non-injected `: Props` after the destructure didn't work - could it be a linting rule or a package version problem?

Comment (jamesgorrie) [reply catalogue/webapp/components/WorkDetailsProperty/WorkDetailsProperty.tsx:7]: I've had the same issue. It's not a type issue, but rather [a linting rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md) issue. 

I personally prefer using the `FunctionComponent` type to infer what the props are and could be, so if we can't find a way to make the linter happy I might have a look at turning the rule off as I'd say static typing trumps linting.

Comment (jamesgorrie) [reply catalogue/webapp/components/WorkDetailsProperty/WorkDetailsProperty.tsx:7]: I'm going to get this in i nit's current state, and we can discuss as a team later.

**Review by jamesgorrie — APPROVED** [review]
For future reference, could we drop a screen grab in as it helps the other non-=dev members of the team get a gist of what the PR is about.

Nice work on this and all the typing!

**Pull request merged**